src/gtk/dnd.cpp

```diff
@@ -806,7 +806,7 @@ void wxDropSource::PrepareIcon( int action, GdkDragContext *context )
         cairo_region_destroy(region);
     }
 
-    g_signal_connect(widget, "draw", G_CALLBACK(draw_icon), icon);
+    g_signal_connect(m_iconWindow, "draw", G_CALLBACK(draw_icon), icon);
```

Comment:
I find it a bit suspicious that we connect to "draw" on the parent of the drawing area widget and not the widget itself now, are you sure this is correct?